e2e/node_config.go

```diff
@@ -448,13 +449,16 @@ func baseTemplateLinux(location string, k8sVersion string) *datamodel.NodeBootst
 		SSHStatus:                 0,
 		DisableCustomData:         false,
 	}
+	config, err := pruneKubeletConfig(k8sVersion, config)
+	require.NoError(t, err)
+	return config
 }
 
 // this been crafted with a lot of trial and pain, some values are not needed, but it takes a lot of time to figure out which ones.
 // and we hope to move on to a different config, so I don't want to invest any more time in this-
-func baseTemplateWindows(location string) *datamodel.NodeBootstrappingConfiguration {
+func baseTemplateWindows(t *testing.T, location string) *datamodel.NodeBootstrappingConfiguration {
 	kubernetesVersion := "1.29.9"
```

Comment:
we should probably have this version also be resolved dynamically from the e2e cluster, out of scope for this PR though